p2p/swarmhandlers.go

```diff
@@ -23,7 +23,7 @@ import (
 // Handles a local request to register a remote node in the swarm
 // Register adds info about this node but doesn't attempt to connect to it
 func (s *swarmImpl) onRegisterNodeRequest(n node.RemoteNodeData) {
-
+	s.peerMapMutex.Lock()
```

Comment:
Since there are places where we only query the peers map you should consider using RWMutex 
[https://golang.org/src/sync/rwmutex.go](url)